src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -725,7 +747,13 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
         {
             if (factoryContext.JsonRequestBodyType is not null)
             {
-                throw new InvalidOperationException("Action cannot have more than one FromBody attribute.");
+                factoryContext.HasAnotherBodyParameter = true;
+                var parameterName = parameter.Name;
+                if (parameterName is not null && factoryContext.TrackedParameters.ContainsKey(parameterName))
```

Comment:
`CreateArgument` already checks `parameterName is not null` so you can just use `parameter.Name!` everywhere and not check if it's null here.

Also, in what circumstance does would `TrackedParameters` *not* contain `parameter.Name!` here? It's always added right before the call to `BindParameterFromBody`, right?

Reply:
I missed cleaning this part. will fix.